examples/gno.land/r/gnoland/hof/hof.gno

```diff
@@ -0,0 +1,174 @@
+// Package hof is the gno.land hall of fame realm.
+// It contains a permanent exhibition of items in the hall of fame,
+// a possibility to host temporary ones created by its admins.
+package hof
+
+import (
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	exhibCounter seqid.ID
+	exhibitions  *avl.Tree // id > *Exhibition
+)
+
+type (
+	Exhibition struct {
+		id          seqid.ID // fixed per exhibition
+		itemCounter seqid.ID
+		title       string
+		description string
+		startTime   *time.Time // given in RFC3339
+		endTime     *time.Time // end time of the event, given in RFC3339
+		items       *avl.Tree  // Item id > Item
+	}
+
+	Item struct {
+		id       seqid.ID
+		exhibID  seqid.ID
+		pkgpath  string
+		deployer std.Address
+		blockNum int64
+		upvote   *avl.Tree // std.Addr > struct{}{}
+		downvote *avl.Tree // std.Addr > struct{}{}
+	}
+)
+
+func init() {
+	exhibitions = avl.NewTree()
+
+	exhibitions.Set(seqid.ID(0).String(),
+		&Exhibition{ // permanent exhibition, ID=0
+			id:    seqid.ID(0),
+			items: avl.NewTree(),
+			title: "Permanent Exhibition",
+		})
+}
+
+// CreateTemporaryExhibition creates a temporary exhibition to be displayed while it's active
+func CreateTemporaryExhibition(title, description string, start, end *time.Time) error {
```

Comment:
Maybe you can turn this multi-expo hall into a second demo while keeping the permanent collection as a straightforward mono-exhibit contract. I'm currently looking for demos that are as simple as possible.

Reply:
[19f7fa9](https://github.com/gnolang/gno/pull/2842/commits/19f7fa93995e3ca2352977708df3c38d2252bf7f)